routes/common/postgres_helper.rb

```diff
@@ -0,0 +1,235 @@
+# frozen_string_literal: true
+
+class Routes::Common::PostgresHelper < Routes::Common::Base
+  def list
+    if @mode == AppMode::API
+      dataset = project.postgres_resources_dataset
+      dataset = dataset.where(location: @location) if @location
+      result = dataset.authorized(@user.id, "Postgres:view").eager(:semaphores, :strand).paginated_result(
+        start_after: @request.params["start_after"],
+        page_size: @request.params["page_size"],
+        order_column: @request.params["order_column"]
+      )
+
+      {
+        items: Serializers::Postgres.serialize(result[:records]),
+        count: result[:count]
+      }
+    else
+      postgres_databases = Serializers::Postgres.serialize(project.postgres_resources_dataset.authorized(@user.id, "Postgres:view").eager(:semaphores, :strand, :representative_server, :timeline).all, {include_path: true})
+      @app.instance_variable_set(:@postgres_databases, postgres_databases)
+      @app.view "postgres/index"
+    end
+  end
+
+  def post(name: nil)
```

Comment:
I'm trying to keep the method name and the method in the route _more or less_ similar, because who knows maybe I would do another pass in a different PR to programmatically call the correct method with some Ruby meta programming :smirk: